ports/szip/portfile.cmake

```
@@ -1,30 +1,63 @@
+set(SZIP_VERSION "2.1.1")
 vcpkg_download_distfile(ARCHIVE
-    URLS "https://support.hdfgroup.org/ftp/lib-external/szip/2.1.1/src/szip-2.1.1.tar.gz"
-    FILENAME "szip-2.1.1.tar.gz"
+    URLS "https://support.hdfgroup.org/ftp/lib-external/szip/${SZIP_VERSION}/src/szip-${SZIP_VERSION}.tar.gz"
+    FILENAME "szip-${SZIP_VERSION}.tar.gz"
     SHA512 ada6406efb096cd8a2daf8f9217fe9111a96dcae87e29d1c31f58ddd2ad2aa7bac03f23c7205dc9360f3b62d259461759330c7189ef0c2fe559704b1ea9d40dd
 )
 
 vcpkg_extract_source_archive_ex(
     OUT_SOURCE_PATH SOURCE_PATH
-    ARCHIVE ${ARCHIVE}
-    REF szip-2.1.1
+    ARCHIVE "${ARCHIVE}"
+    REF szip-${SZIP_VERSION}
     PATCHES
-        fix-szip-config-to-set-szip-found.patch # This patch is required for linux on osx; It does not matter for windows
+        fix-linkage-config.patch
+        mingw-lib-names.patch
 )
 
-vcpkg_configure_cmake(
-    SOURCE_PATH ${SOURCE_PATH}
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
     PREFER_NINJA
     OPTIONS
         -DSZIP_INSTALL_DATA_DIR=share/szip/data
         -DSZIP_INSTALL_CMAKE_DIR=share/szip
 )
 
-vcpkg_install_cmake()
+vcpkg_cmake_install()
 vcpkg_copy_pdbs()
-vcpkg_fixup_cmake_targets()
+vcpkg_cmake_config_fixup()
 
-file(RENAME ${CURRENT_PACKAGES_DIR}/share/szip/data/COPYING ${CURRENT_PACKAGES_DIR}/share/szip/copyright)
+if(VCPKG_LIBRARY_LINKAGE STREQUAL "dynamic")
+    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/szip_adpt.h"
+        "\n#ifdef SZ_BUILT_AS_DYNAMIC_LIB"
+        "\n#ifndef SZ_BUILT_AS_DYNAMIC_LIB\n#define SZ_BUILT_AS_DYNAMIC_LIB 1\n#endif\n#ifdef SZ_BUILT_AS_DYNAMIC_LIB")
```

```suggestion
    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/szip_adpt.h"
        "\n#ifdef SZ_BUILT_AS_DYNAMIC_LIB"
        "\n#if 1")
```
It's much cleaner this way (unless there are other places that use this macro)

---

> unless there are other places that use this macro

But I cannot know if consuming code relies on this macro. That's why I decided against the simple define.

OTOH szip is consumed only via hdf5. The macro is in the cmake config files, and I can put it into the Cflags of the pc file which I added in this port (and Require in hdf5 pc file).

---

Applied simpler replacement, and added flag to pc file for dynamic build.